cmake/scripts/common/AddonHelpers.cmake

```diff
@@ -120,18 +120,17 @@ macro (build_addon target prefix libs)
             foreach(src_file ${USED_SOURCES})
               file(STRINGS ${src_file} BIN_ADDON_SRC_PARTS)
               foreach(loop_var ${BIN_ADDON_SRC_PARTS})
-                string(FIND "${loop_var}" "#include" matchres)
-                if("${matchres}" EQUAL 0)
-                  string(REPLACE " " ";" loop_var "${loop_var}")
-                  list(GET loop_var 1 include_name)
-                  string(REGEX REPLACE "[<>\"]|kodi/" "" include_name "${include_name}")
-                  if(include_name MATCHES ${depend_header})
+                string(REGEX MATCH "^[ \t]*#[ \t]*(include|import)[ \t]*[\"<].*[\">]" loop_var "${loop_var}")
+                if(loop_var AND loop_var MATCHES "${depend_header}") # WARNING: Temporary until PVR and Game addons are changed to new C++ style
```

Comment:
This is wrong, because it doesn't check for an exact match inside `"..."` or `<...>`.

Reply:
I change but there comes also another problem with `if(NOT used_type_name IN_LIST CURRENT_DEFS)`

e.g. pvr.demo still declare: `cmake_minimum_required(VERSION 2.6)`
where generates the `Policy CMP0057 is not set:` where it need `3.3`.

Before must all the PVR CMakeLists.txt updated
Like you already sayed :smile::
> Keep in mind that if(<variable|string> IN_LIST <variable>) requires every add-on to have at least cmake_minimum_required(VERSION 3.3) set in it's main CMakeLists.txt.

I think then to continue my huge addon update chain, now also to PVR. Before I continue here.